### DISCUSSION TITLE
TypeScript?

### DISCUSSION THREAD

**Discussion by KevinBurton** [vercel/next.js#18255] [Help]
I am very new to NextJS so I am trying to follow the learn sample. But when I do something like `'touch tsconfig.json'` I get the error:

```
ready - started server on http://localhost:3000
Error: Line 1: Unexpected token ILLEGAL
```

It was my understanding that creating a blank tsconfig.json file was how I "enabled" TypeScript?

**Comment by jamesmosier** [comment]
Hi @KevinBurton. Assuming you have no installed these dependencies, running `next` (via `npm run dev` or `yarn dev`)  should have prompted you to install these dependencies too:

```
yarn add --dev typescript @types/react @types/node
```

or

```
npm install typescript @types/react @types/node --save-dev
```

**Reply by jamesmosier** [reply]
> It did. So I ran the suggested package installations. But I still get the
error message when I ‘npm run dev’.

@KevinBurton Do you have a more complete repository/reproduction as well as a more complete error log?

**Comment by rphuber** [comment]
@KevinBurton that is exactly how I was able to set up typescript.  I just added a `tsconfig.json` file.  The next time I tried to start the dev server it gave me the prompt that @jamesmosier mentioned.  After adding those packages it filled in my `tsconfig.json` file for me automatically.  Super easy.

**Comment by KevinBurton** [comment]
It did. So I ran the suggested package installations. But I still get the
error message when I ‘npm run dev’.

**Comment by KevinBurton** [comment]
Here are the steps that I take on another computer:

npx create-next-app nextjs-blog --use-npm --example "
> https://github.com/vercel/next-learn-starter/tree/master/learn-starter"
> cd nextjs-blog
> touch tsconfig.json
> yarn add --dev typescript @types/react @types/node
> npm run dev


For this computer I get this response (in the logs):

0 verbose cli [
> 0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
> 0 verbose cli   'C:\\Program
> Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
> 0 verbose cli   'run',
> 0 verbose cli   'dev'
> 0 verbose cli ]
> 1 info using npm@7.0.3
> 2 info using node@v15.0.1
> 3 timing config:load:defaults Completed in 2ms
> 4 timing
> config:load:file:C:\Users\RKevi\AppData\Roaming\nvm\v15.0.1\node_modules\npm\npmrc
> Completed in 1ms
> 5 timing config:load:builtin Completed in 1ms
> 6 timing config:load:cli Completed in 3ms
> 7 timing config:load:env Completed in 0ms
> 8 timing config:load:project Completed in 4ms
> 9 timing config:load:file:C:\Users\RKevi\.npmrc Completed in 2ms
> 10 timing config:load:user Completed in 2ms
> 11 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in
> 0ms
> 12 timing config:load:global Completed in 0ms
> 13 timing config:load:cafile Completed in 0ms
> 14 timing config:load:validate Completed in 0ms
> 15 timing config:load:setUserAgent Completed in 0ms
> 16 timing config:load:setEnvs Completed in 1ms
> 17 timing config:load Completed in 15ms
> 18 verbose npm-session 7164990d57ef68ff
> 19 timing npm:load Completed in 27ms
> 20 timing command:run-script Completed in 2ms
> 21 verbose stack Error: missing script: dev
> 21 verbose stack     at runScript
> (C:\Users\RKevi\AppData\Roaming\nvm\v15.0.1\node_modules\npm\lib\run-script.js:53:11)
> 22 verbose cwd C:\Users\RKevi\source\repos
> 23 verbose Windows_NT 10.0.19042
> 24 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program
> Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "dev"
> 25 verbose node v15.0.1
> 26 verbose npm  v7.0.3
> 27 error missing script: dev
> 28 verbose exit 1

**Comment by KevinBurton** [comment]
Sorry. I wasn't in the right folder. Doing the same steps that I outlined
earlier I get this output:

PS C:\Users\RKevi\source\repos\nextjs-blog> npm run dev
> > learn-starter@0.1.0 dev
> > next dev
> ready - started server on http://localhost:3000
> Error: Line 1: Unexpected token ILLEGAL
>     at ErrorHandler.constructError
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:123955)
>     at ErrorHandler.createError
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:124191)
>     at ErrorHandler.throwError
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:124325)
>     at Scanner.throwUnexpectedToken
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:129084)
>     at Scanner.scanPunctuator
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:137130)
>     at Scanner.lex
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:146217)
>     at Tokenizer.getNextToken
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:151607)
>     at Object.tokenize
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:8626)
>     at p
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:4129)
>     at Object.et [as parse]
> (C:\Users\RKevi\source\repos\nextjs-blog\node_modules\next\dist\compiled\comment-json\index.js:1:6267)
> {
>   index: 0,
>   lineNumber: 1,
>   description: 'Unexpected token ILLEGAL'
> }
> npm ERR! code 1
> npm ERR! path C:\Users\RKevi\source\repos\nextjs-blog
> npm ERR! command failed
> npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c "next dev"
> npm ERR! A complete log of this run can be found in:
> npm ERR!
> C:\Users\RKevi\AppData\Local\npm-cache\_logs\2020-10-27T01_09_48_865Z-debug.log


The log looks like:

0 verbose cli [
> 0 verbose cli   'C:\\Program Files\\nodejs\\node.exe',
> 0 verbose cli   'C:\\Program
> Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
> 0 verbose cli   'run',
> 0 verbose cli   'dev'
> 0 verbose cli ]
> 1 info using npm@7.0.3
> 2 info using node@v15.0.1
> 3 timing config:load:defaults Completed in 2ms
> 4 timing
> config:load:file:C:\Users\RKevi\AppData\Roaming\nvm\v15.0.1\node_modules\npm\npmrc
> Completed in 1ms
> 5 timing config:load:builtin Completed in 1ms
> 6 timing config:load:cli Completed in 3ms
> 7 timing config:load:env Completed in 1ms
> 8 timing config:load:file:C:\Users\RKevi\source\repos\nextjs-blog\.npmrc
> Completed in 2ms
> 9 timing config:load:project Completed in 3ms
> 10 timing config:load:file:C:\Users\RKevi\.npmrc Completed in 2ms
> 11 timing config:load:user Completed in 2ms
> 12 timing config:load:file:C:\Program Files\nodejs\etc\npmrc Completed in
> 0ms
> 13 timing config:load:global Completed in 1ms
> 14 timing config:load:cafile Completed in 0ms
> 15 timing config:load:validate Completed in 0ms
> 16 timing config:load:setUserAgent Completed in 1ms
> 17 timing config:load:setEnvs Completed in 0ms
> 18 timing config:load Completed in 14ms
> 19 verbose npm-session ea5eadeb75d9a59d
> 20 timing npm:load Completed in 24ms
> 21 timing command:run-script Completed in 1874ms
> 22 verbose stack Error: command failed
> 22 verbose stack     at ChildProcess.<anonymous>
> (C:\Users\RKevi\AppData\Roaming\nvm\v15.0.1\node_modules\npm\node_modules\@npmcli\promise-spawn\index.js:63:27)
> 22 verbose stack     at ChildProcess.emit (node:events:327:20)
> 22 verbose stack     at maybeClose (node:internal/child_process:1048:16)
> 22 verbose stack     at Process.ChildProcess._handle.onexit
> (node:internal/child_process:288:5)
> 23 verbose pkgid learn-starter@0.1.0
> 24 verbose cwd C:\Users\RKevi\source\repos\nextjs-blog
> 25 verbose Windows_NT 10.0.19042
> 26 verbose argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program
> Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "dev"
> 27 verbose node v15.0.1
> 28 verbose npm  v7.0.3
> 29 error code 1
> 30 error path C:\Users\RKevi\source\repos\nextjs-blog
> 31 error command failed
> 32 error command C:\WINDOWS\system32\cmd.exe /d /s /c "next dev"
> 33 verbose exit 1

**Comment by rafaelalmeidatk** [comment]
Delete the `tsconfig.json` file, you are receiving an error because there is no content in the file. Next.js will generate it for you when you start a project that has TS files for the first time

**Comment by KevinBurton** [comment]
So the instructions at
https://nextjs.org/learn/excel/typescript/create-tsconfig no longer apply?
